app/services/ui_notifications/clean_expired.rb

```diff
@@ -0,0 +1,41 @@
+module UINotifications
+  class CleanExpired
+    def initialize(options = {})
+      if options[:blueprint].present?
+        blueprint = NotificationBlueprint.find_by(name: options[:blueprint])
+        raise(Foreman::Exception, 'blueprint name does not exist') if blueprint.nil?
+        @blueprint = blueprint.name
+      end
+      parsed_time = options[:expired_at].is_a?(String) ? options[:expired_at].to_time : Time.now.utc
+      raise(Foreman::Exception, 'Expired time cannot be greater then now') if parsed_time > Time.now.utc
+      @expired_at = parsed_time
+      @group = options[:group]
+    end
+
+    def clean!
+      UINotifications::Base.logger.info("Removing all expired notifications")
+      to_delete = Notification.joins(:notification_blueprint)
+                      .where('notifications.expired_at < ?', @expired_at)
+                      .where(blueprint_query)
+                      .where(group_query)
+      NotificationRecipient.where(notification_id: to_delete.map {|n| n.id }).delete_all
+      @deleted_count = to_delete.delete_all
```

Comment:
The general idea I took from [Fact_cleaner](https://github.com/theforeman/foreman/blob/develop/app/services/fact_cleaner.rb#L13), It seems  better then just return a deleted count, it stored for later use if needed. (I don't quite sure if in the future clean notifications will occur only as rake task).